src/contract/Wallet.ts

```diff
@@ -0,0 +1,42 @@
+import Transaction from 'arweave/node/lib/transaction';
+import { Warp } from 'core/Warp';
+import { Signature, SigningFunction } from './Contract';
+import { ArWallet } from './deploy/CreateContract';
+
+export class Wallet {
+  public signature: Signature;
+  private readonly warp: Warp;
+
+  constructor(warp: Warp) {
+    this.warp = warp;
+  }
+
+  getSignature(signature: ArWallet | Signature) {
```

Comment:
it would be awesome to add some tests to this wicked logic (especially since we now have it extracted to a separate class).